tasks/createTagsTasks.ts

```diff
@@ -0,0 +1,169 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as gulp from 'gulp';
+import * as xml2js from 'xml2js';
+import * as fs from 'fs';
+import axios, { AxiosResponse } from 'axios';
+import * as minimist from 'minimist';
+import { Octokit } from '@octokit/rest';
+
+interface Options {
+    releaseVersion: string;
+    releaseCommit: string;
+}
+
+gulp.task('createTags', async (): Promise<void> => {
+    const options = minimist<Options>(process.argv.slice(2));
+    console.log(`releaseVersion: ${options.releaseVersion}`);
+    console.log(`releaseCommit: ${options.releaseCommit}`);
+
+    const roslynCommit = await findRoslynCommitAsync();
+    if (!roslynCommit) {
+        logError('Failed to find roslyn commit.');
+        return;
+    }
+
+    const tagCreatedInRoslyn = await tagRepoAsync(
+        'dotnet',
+        'roslyn',
+        roslynCommit,
+        `VSCode-CSharp-${options.releaseVersion}`,
+        `${options.releaseVersion} VSCode C# extension release`
+    );
+
+    if (!tagCreatedInRoslyn) {
+        logError('Failed to tag roslyn');
+        return;
+    }
+
+    console.log('tag created in roslyn');
+
+    const tagCreatedInVsCodeCsharp = await tagRepoAsync(
+        'dotnet',
+        'vscode-csharp',
+        options.releaseCommit,
+        options.releaseVersion,
+        options.releaseVersion
+    );
+
+    if (!tagCreatedInVsCodeCsharp) {
+        logError('Failed to tag vscode-csharp');
+        return;
+    }
+
+    console.log('tag created in vscode-csharp');
+});
+
+async function findRoslynCommitAsync(): Promise<string | null> {
+    const packageJsonString = fs.readFileSync('./package.json').toString();
+    const packageJson = JSON.parse(packageJsonString);
+    const roslynVersion = packageJson['defaults']['roslyn'];
+    if (!roslynVersion) {
+        logError("Can't find roslyn version in package.json");
+        return null;
+    }
+
+    console.log(`Roslyn version is ${roslynVersion}`);
+    const nugetFileString = fs.readFileSync('./server/NuGet.config').toString();
+    const nugetConfig = await xml2js.parseStringPromise(nugetFileString);
+    const nugetServerIndex = nugetConfig.configuration.packageSources[0].add[0].$.value;
+    if (!nugetServerIndex) {
+        logError("Can't find nuget server index.");
+        return null;
+    }
+
+    console.log(`Nuget server index is ${nugetServerIndex}`);
+    // Find the matching commit from nuget server
+    // 1. Ask the PackageBaseAddress from nuget server index.
```

Comment:
How it works:
1. Parse the package.json to get roslyn version.
2. Parse the nuget.config to get server index.
3. Send http request to nuget server index, it would gives something like this:
<img width="923" alt="image" src="https://github.com/dotnet/vscode-csharp/assets/24360909/6fa4486e-eea1-41d3-a088-e45363bcd160">
This is required to be implemented by nuget server so we should be able to get it.
4. Use that to get nuspec file https://learn.microsoft.com/en-us/nuget/api/package-base-address-resource#download-package-manifest-nuspec
5. From nuspec file, find the commit associated with the nuget package.

Reply:
So web requests are kind of flaky and I'm hesitent to have the release CI fail because of them.  

I think a more reliable approach would be to write out the commit to some file in the .roslyn directory when we download the package.  Because when we download the package we have the .nuspec file on disk already - we can just read it and extract the commit to a file inside the .roslyn directory.
<img width="559" alt="image" src="https://github.com/dotnet/vscode-csharp/assets/5749229/8851b30d-5ac7-436b-89bd-efa88501d972">

Reply:
Fixed